opengrok-indexer/src/main/java/org/opengrok/indexer/history/FileAnnotationCache.java

```diff
@@ -169,11 +171,16 @@ public Annotation get(File file, @Nullable String rev) throws CacheException {
              * should be present to catch weird cases of someone not using the store() or general badness.
              */
             if (storedRevision == null) {
-                LOGGER.log(Level.FINER, "no stored revision in annotation cache for ''{0}''", file);
+                if (LOGGER.isLoggable(Level.FINER)) {
```

Comment:
We can use logger with msg supplier parameter to avoid this if condition. Having supplier type(lambda expression) parameter will make sure that the parameter is only evaluated if corresponding log level is enabled
Ex: LOGGER.finer( () -> String.format("no stored revision in annotation cache for ''%s''",launderLog(file.toString())));

Reply:
Good idea, will be more readable that way.